doc/explanation/components/reactive_html_components.md

```diff
@@ -1,12 +1,171 @@
 
-The `ReactiveHTML` provides bi-directional syncing of arbitrary HTML attributes and DOM properties with parameters on the subclass. This kind of component must declare a HTML template written using Javascript template variables (`${}`) and optionally Jinja2 syntax:
+## Why Use ReactiveHTML?
+
+`ReactiveHTML` empowers you to design and build custom components that seamlessly integrate with your Panel applications. These components can enhance your applications' interactivity and functionality, all while keeping the development process straightforward and free from the complexities of JavaScript build tools.
+
+## What is a `ReactiveHTML` component?
+
+A *`ReactiveHTML` component* is essentially a class that you create by inheriting from the `ReactiveHTML` class, much like the example [shown below](#what-does-a-basic-example-look-like). Within this custom class, you are required to define the `_template` attribute using HTML, which serves as the *design blueprint* for your custom component. You can use [Jinja2](https://jinja.palletsprojects.com) syntax as well as *template variables* `${...}` to make the template *dynamic*.
+
+## What does a basic example look like?
+
+A basic `ReactiveHTML` component looks like below
+
+```{pyodide}
+import param
+from panel.reactive import ReactiveHTML
+import panel as pn
+pn.extension() # for notebook
+
+class Slideshow(ReactiveHTML):
+
+    index = param.Integer(default=0)
+
+    _template = '<img id="slideshow" src="https://picsum.photos/800/300?image=${index}" onclick="${_img_click}"></img>'
+
+    def _img_click(self, event):
+        self.index += 1
+
+Slideshow(width=800, height=300)
+```
+
+## Where do I find practical examples?
+
+To see `ReactiveHTML` in action and discover **how to** create your custom components, check out our detailed guide: [How-to > Create Custom Components with ReactiveHTML](../../how_to/custom_components/reactive_html/index.md). It's packed with practical examples to help you get started quickly.
+
+## What does the API of `ReactiveHTML` look like?
+
+It looks like described [here](../../api/panel.reactive.html#panel.reactive.ReactiveHTML).
+
+### What class attributes does `ReactiveHTML` expose?
+
+#### `_template` (str)
+
+The *blueprint* for how your custom component should look and behave. The *_template* is defined using HTML. You can use [Jinja2](https://jinja.palletsprojects.com) syntax as well as *template variables* `${...}` to make the template *dynamic*.
+
+### `_child_config` (Optional[dict])**
+
+Controls how template variables ${...} will be rendered when inserted as children into the HTML
+elements
+
+- `model` (default): Create child and render as Panel component.
+- `literal`: Create child and insert the string value as `innerHTML`
+- `template`: Create child and insert the string value as `innerText`
```

Comment:
I need help to define these @philippjfr . I've seen different definitions and they are not consistently aligned with what I experience when using them. See for example examples and questions below. I'm so confused.

- Can you confirm the are all rendered on `js` side. Sometimes I'm thinking either `literal` or `template` is rendered on python side using Jinaj2.
- I've tried to look at the .py and .ts code many times. But I cannot find the logic that distinguishes between `literal` and `template`. So I cannot learn from the code. Where is that code?
- What does the names `literal` and `template` refer too? Please give me something so that I can remember the difference.
- Why does the *escaping* not work consistently for `literal` and `template` values? Is it a feature or a bug? It makes it so hard to understand and trust that this is robust. See examples below.
- Why can I set the `v_model` parameter value below to a markdown string that is rendered as a Markdown pane. But if I try setting it to an `svg` string then I get an error because a `String` parameter cannot be assigned a `SVG` pane? Is is a bug or a feature. I'm so confused here.
- Why does `ReactiveHTML` even try to change parameter values to their Panel pane? It makes it so hard to use `DataFrame` etc. parameters.

Reply:
This is the example that confuses me so much

```python
import panel as pn
import param
from panel.reactive import ReactiveHTML

pn.extension()

class CustomComponent(ReactiveHTML):
    v_model = param.String(default="I'm a **model** value. <em>emphasize</em>")
    v_literal = param.String(default="I'm a **literal** value. <em>emphasize</em>")
    v_template = param.String(default="I'm a **template** value. <em>emphasize</em>")

    _child_config = {
        "v_model": "model",
        "v_literal": "literal",
        "v_template": "template",
    }

    _template = """
<div id="el_model">${v_model}</div>
<div id="el_literal">${v_literal}</div>
<div id="el_template">${v_template}</div>
"""


component = CustomComponent(width=500, height=200).servable()
component.v_model=component.v_model.replace("**", "*")
component.v_literal=component.v_model.replace("**", "*")
component.v_template=component.v_model.replace("**", "*")
print(component.v_model)

svg = """<svg style="stroke: #e62f63;" width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg" slot="collapsed-icon">
<path d="M15.2222 1H2.77778C1.79594 1 1 1.79594 1 2.77778V15.2222C1 16.2041 1.79594 17 2.77778 17H15.2222C16.2041 17 17 16.2041 17 15.2222V2.77778C17 1.79594 16.2041 1 15.2222 1Z" stroke-linecap="round" stroke-linejoin="round"></path>
<path d="M9 5.44446V12.5556" stroke-linecap="round" stroke-linejoin="round"></path>
<path d="M5.44446 9H12.5556" stroke-linecap="round" stroke-linejoin="round"></path></svg>"""
CustomComponent(v_literal=svg, v_template=svg, width=500, height=200).servable()
```

![image](https://github.com/holoviz/panel/assets/42288570/a9b859bf-96ee-4883-935d-64d0b1a0773d)